### PR TITLE
Homework 2

### PR DESCRIPTION
**Pull request by avozniak**

### CODE REVIEW

**Review by oleksandr-kychun — COMMENTED** [review]
Interesting) correct object_model implementation.
about state_and_behavior:
prefer using getter/setter instead of instance variables
it is better to provide defaults via hsh.fetch method
it is weird though to see the same defaults for the fourth time at this point)

File: lib/topics/object_model.rb [hunk 1]
```diff
@@ -2,4 +2,24 @@
 
 # exercise object model
 module ObjectModel
+  # Define Linux Friendly module
+  module LinuxFriendly
+    def fork_process
+      super
+    end
+  end
+  # Define Desktop class
```
Comment (vbyno) [inline lib/topics/object_model.rb]: empty lines are needed between logical parts. New class/module is a new logical part of an application

File: lib/topics/state_and_behaviour.rb [hunk 2]
```diff
@@ -2,4 +2,26 @@
...
+
+    def push_break(speed)
+      speed > @current_speed ? @current_speed : @current_speed -= speed
+    end
+
+    def speed_up(speed)
+      @current_speed += speed
+    end
+
+    def self.default_car
```
Comment (vbyno) [inline lib/topics/state_and_behaviour.rb]: please, read the style guide (incorrect methods order)

File: lib/topics/state_and_behaviour.rb [hunk 3]
```diff
@@ -2,4 +2,26 @@
 
 # exercise state and behaviour
 module StateAndBehaviour
+  # Define new class Car
+  class Car
+    attr_accessor :year, :color, :model, :current_speed
+    def initialize(args)
```
Comment (vbyno) [inline lib/topics/state_and_behaviour.rb]: empty line is needed

File: lib/topics/state_and_behaviour.rb [hunk 4]
```diff
@@ -2,4 +2,26 @@
...
+      @model = args[:model] || 'skyline'
+      @current_speed = 0
+    end
+
+    def push_break(speed)
+      speed > @current_speed ? @current_speed : @current_speed -= speed
+    end
+
+    def speed_up(speed)
+      @current_speed += speed
```
Comment (vbyno) [inline lib/topics/state_and_behaviour.rb:30]: why don't you use setter here?

File: lib/topics/state_and_behaviour.rb [hunk 5]
```diff
@@ -2,4 +2,26 @@
...
+    end
+
+    def self.default_car
+      new
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb]: and new will fail. because initializer requires one argument.

File: lib/topics/object_model.rb (hunk 1 above)
Comment (avozniak) [reply lib/topics/object_model.rb]: done

File: lib/topics/state_and_behaviour.rb (hunk 2 above)
Comment (avozniak) [reply lib/topics/state_and_behaviour.rb]: done

File: lib/topics/state_and_behaviour.rb (hunk 3 above)
Comment (avozniak) [reply lib/topics/state_and_behaviour.rb]: add an empty line

File: lib/topics/state_and_behaviour.rb (hunk 4 above)
Comment (avozniak) [reply lib/topics/state_and_behaviour.rb:30]: you mean setter for speed or something else?

File: lib/topics/state_and_behaviour.rb [hunk 6]
```diff
@@ -2,4 +2,27 @@
...
+  # Define new class Car
+  class Car
+    attr_reader :year, :color, :model, :current_speed
+
+    def self.default_car
+      Car.new({})
+    end
+
+    def initialize(args)
+      @year  = args[:year]  || 2015
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb]: 2015 magic number

Comment (avozniak) [reply lib/topics/state_and_behaviour.rb]: hope I understand your words right

**Pull request closed without merging**